src/Rewards.sol

```diff
@@ -94,6 +94,12 @@ contract Rewards is AccessControl, EIP712 {
      * @dev The sequence number of the batch reward.
      */
     uint256 public batchSequence;
+    /**
+     * @dev Determines the amount of rewards to be minted for the submitter of batch as a percentage of total rewards in that batch.
+     * This value indicates the cost overhead of minting rewards that the network is happy to take. Though it is set to 2% by default,
+     * the governance can change it to ensure the network is sustainable.
+     */
+    uint256 public batchSubmitterRewardPercentage = 2;
```

Comment:
Do you want this to be a constructor parameter?

Reply:
It needs to remain changeable anyway but we can definitely allow the constructor to set it too.

Reply:
I think it is cleaner yeah

Reply:
Just fixed it as suggested